tests/trogdor/trogdor_utils.py

```diff
@@ -0,0 +1,202 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import array
+import re
+import sys
+import json
+
+from confluent_kafka.cimpl import KafkaException, KafkaError, Producer
+
+from confluent_kafka.admin import AdminClient, NewTopic
+
+
+def partition_set(topics):
+    """
+    Return an array of TopicPartition set, so we can send messages to them one by one.
+    For example, for topics
+    {
+      "foo" : { "numPartitions": 3 }
+      "bar" : { "numPartitions": 2 }
+    }
+    return
+    [("foo", 0), ("foo", 1), ("foo", 2), ("bar", 0), ("bar", 1)]
+    """
+    topic_partitions = []
+    for t_name in topics:
+        t = topics[t_name]
+        nr_partitions = t.get("numPartitions", 1)
+        for i in range(0, nr_partitions):
+            topic_partitions.append((t_name, i))
+    return topic_partitions
+
+
+def expand_topics(topics):
+    """
+    Return extended topics.
+    For example, for topics { "foo[1-3]" : {} }, return { "foo1" : {}, "foo2" : {}, "foo3" : {}}
+    """
+    topic_expand_matter = re.compile(r'(.*?)\[([0-9]*)\-([0-9]*)\](.*)')
+    expanded = {}
+    for topicName in topics:
+        match = topic_expand_matter.match(topicName)
+        if match:
+            pre = match.group(1)
+            start = int(match.group(2))
+            end = int(match.group(3))
+            last = match.group(4)
+            if end >= start:
+                for t in range(start, end + 1):
+                    newTopicName = "%s%d%s" % (pre, t, last)
+                    expanded[newTopicName] = topics[topicName]
+            else:
+                raise Exception('Invalid range in the topic name %s' % (topicName))
+        else:
+            expanded[topicName] = topics[topicName]
+    return expanded
+
+
+def merge_topics(active_topics, inactive_topics):
+    all_topics = active_topics.copy()
+    all_topics.update(inactive_topics)
+    return all_topics
+
+
+def create_kafka_conf(bootstrap_servers, *args):
+    k_conf = {"bootstrap.servers": bootstrap_servers}
+    for conf in args:
+        k_conf.update(conf)
+    return k_conf
+
+
+def create_admin_conf(bootstrap_servers, common_client_config, admin_client_config):
+    # Refer Java Trogdor tools/src/main/java/org/apache/kafka/trogdor/common/WorkerUtils.java#L305
+    admin_request_timeout_ms = 25000
+    create_kafka_conf(bootstrap_servers, common_client_config, admin_client_config)
+    admin_conf = create_kafka_conf(bootstrap_servers, common_client_config, admin_client_config)
+    admin_conf["socket.timeout.ms"] = admin_request_timeout_ms
```

Comment:
Why is this needed? The defaults should be good.

Reply:
Try to be same as the Java Trogdor produce worker.

Reply:
Okay, let's avoid setting specific configs unless we know it is needed.

Reply:
Please remove the unneeded configuration.

Reply:
@edenhill Done.